src/EnergyPlus/api/api.py

```diff
@@ -0,0 +1,94 @@
+from ctypes import cdll, c_char_p
+import os
+import sys
+
+from pyenergyplus.func import Functional
+from pyenergyplus.datatransfer import DataExchange
+from pyenergyplus.runtime import Runtime
+
+
+def api_path() -> str:
+    """
+    This function returns a string to the EnergyPlus dynamic library.  The energyplusapi target in the build system
+    depends on the Python API build script, so you shouldn't be able to generate the EnergyPlus dynamic library without
+    these scripts being successfully set up in the build tree by CMake.
+
+    :return: A string absolute path to the EnergyPlus DLL.
+    """
+    this_script_dir = os.path.dirname(os.path.realpath(__file__))
+    api_dll_dir = os.path.dirname(os.path.normpath(this_script_dir))
+    if sys.platform.startswith('linux'):
+        return os.path.join(api_dll_dir, 'libenergyplusapi.so')
+    elif sys.platform.startswith('darwin'):
+        return os.path.join(api_dll_dir, 'libenergyplusapi.dylib')
+    else:  # assume Windows
+        return os.path.join(api_dll_dir, 'EnergyPlusAPI.dll')
+
+
+class EnergyPlusAPI:
+    """
+    This class exposes the EnergyPlus C Library API to Python.  The API is split into three categories, and this class
+    exposes each API category through member variables.  If an instance of this class is created as
+    `api = EnergyPlusAPI()`, then the following members are available:
+
+    - `api.functional`: The functional API provides access to static API calls, such as thermophysical property methods.
+    - `api.runtime`: The runtime API allows a user to provide Python functions as callbacks, which are then
+      called from within EnergyPlus at specific points in the simulation.
+    - `api.exchange` The data_exchange API allows a user to exchange data (get sensor values, set actuator values)
```

Comment:
I guess I'm not fully understanding the distinction between 'runtime' and 'exchange' here.

Reply:
Yeah, in reality, there doesn't need to be any distinction between any of these three, it's purely an organization thing.  The "runtime API" provides the hooks to interact with a simulation at "runtime".  So those functions allow you to register a python or C callback function to be called during a simulation run.  You register those functions, then call energyplus as a library and you can do whatever you want inside those functions.

One of the things you'll likely want to do in one of those callbacks is get access to E+ data and actuators.  That is what the data exchange API classification is all about -- getting variable and actuator handles, getting variable values and setting actuator values.